lib/Twig/NodeVisitor/Optimizer.php

```diff
@@ -45,6 +46,7 @@ public function __construct($optimizers = -1)
         }
 
         $this->optimizers = $optimizers;
+        $this->isPHP540RCorHigher = version_compare(phpversion(), '5.4.0RC1', '>=');
```

Comment:
I've made the change, hope it is better.
Since PHP_VERSION_ID is available since 5.2.7 and Twig requires >=5.2.4 we cannot rely on it to be defined.